### PR TITLE
Service subscription on invoice payment

### PR DESCRIPTION
**Pull request by fabriceumuhire**
#### what does this PR do?
- This PR adds the payment functionality and service subscription upon payment

#### Description of task to be completed
- Add an endpoint which will update invoice status from pending to paid
- Upon successful payment, service status should be changed to active

#### How should this be manually tested
- Clone the repo `git clone https://github.com/KitechRW/Kitech_backend.git`
- Run `cd Kitech_backend.git`
- Run ` git checkout ft-service-subscription-on-payment`
- Run `npm install`
- Use POSTMAN with `PATCH` method on this URL `http://localhost:3000/api/v1/invoice/:id`
- On successful payment invoice data should be displayed (refer to the screenshot).
#### Any background context you want to provide
- You need to be authenticated
- You role should be a `Manager`

#### What are the relevant trello story link
[Trello story link](https://trello.com/c/fHRkC1dR)
#### Screenshots (if appropriate)
Screenshot from 2021-01-29 12-09-25
- Only managers should update the invoice

#### Questions:

### CODE REVIEW

**Comment by fabriceumuhire** [comment]
[The user should be subscribed to the service once an invoice is paid.](https://trello.com/c/fHRkC1dR/22-the-user-should-be-subscribed-to-the-service-once-an-invoice-is-paid)

**Comment by luc-tuyishime** [comment]
Can you add a a screenshot in the PR description ?

**Comment by fabriceumuhire** [comment]
> Can you add a a screenshot in the PR description ?

@luc-tuyishime I have added screenshot for invoice updated. Kindly check and let me know if anything is missing.

File: src/modules/invoice/invoice.helper.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  static async generatePDF(body) {
+    const html = fs.readFileSync('template.html', 'utf8');
+    const options = { rmat: "A6", orientation: "landscape","footer": {"height": "20mm", contents: '<div style="font-weight: 900; margin-top: 00px; text-align: center;">Hey !! Pay your invoice within 5 days</div>'}
+    }
+    const users = [ { orderId: body.orderId.trim(""), due_date: body.due_date.trim(""), amount: `${body.amount} FRW` } ]
+    const document = { html, data: { users }, path: "./invoice.pdf" };
+
+    const invoice = await pdf.create(document, options);
+    if(invoice.filename) {
+      await sendmail('k.joshua855@gmail.com','Joshua');
```
Comment (luc-tuyishime) [inline src/modules/invoice/invoice.helper.js:31]: Can you explain why this is email is used here and hard coded ?

Comment (fabriceumuhire) [reply src/modules/invoice/invoice.helper.js]: @luc-tuyishime I have made changes as requested. Thanks for the feedback.

**Review by fridolinho — COMMENTED** [review]
Hello @fabriceumuhire please rebase this PR.

**Pull request merged**